teps/0047-pipeline-task-display-name.md

```diff
@@ -0,0 +1,160 @@
+---
+status: proposed
+title: Pipeline Task Display Name
+creation-date: '2021-01-28'
+last-updated: '2021-02-09'
+authors:
+- '@itewk'
+---
+
+# TEP-0047: Pipeline Task Display Name
+---
+
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+  - [Use Cases (optional)](#use-cases-optional)
+- [Requirements](#requirements)
+- [Proposal](#proposal)
+  - [Notes/Caveats (optional)](#notescaveats-optional)
+  - [Risks and Mitigations](#risks-and-mitigations)
+  - [User Experience (optional)](#user-experience-optional)
+  - [Performance (optional)](#performance-optional)
+- [Design Details](#design-details)
+- [Test Plan](#test-plan)
+- [Design Evaluation](#design-evaluation)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+- [Infrastructure Needed (optional)](#infrastructure-needed-optional)
+- [Upgrade &amp; Migration Strategy (optional)](#upgrade--migration-strategy-optional)
+- [References (optional)](#references-optional)
+<!-- /toc -->
+
+## Summary
+
+A task in a Pipeline is currently represented in the UI (as observed in OpenShift 4.6)
+using a field (`name`) that is meant to be machine readable, not human readable.
+There should be the addition of a new optional field (`displayName`) to describe
+Pipeline tasks that are human readable so that any tool that renders Tekton Pipeline
+can do so using a "human readable" field.
+
+## Motivation
+
+The end user of a Tekton pipeline can vary including but not limited to
+application developer, to security professional, to compliance officer,
+service reliability engineer, product manager, and beyond. The farther
+you move away from the application developer the more important human
+understanding of the CI/CD process is important.
+
+Currently if a product manager or auditor was to look at a Tekton Pipeline run rendered in UI
+that would be presented with a bunch of truncated machine readable task names, no spaces,
+no special characters, no capitalization.
+
+Compare this to other tools in this space, Jenkins, GitLab CI, GitHub Actions, etc.
+Their renderings of their workflows are meant to be human consumable. Their task names have
+the option to be rendered in a human consumable format with use of any characters and fully
+rendered for context.
+
+### Goals
+
+* Add a `Pipeline.spec.tasks.*.displayName` (optional) (or similar) field that allows any text and
+  if provided will be useable by UIs rendering Tekton Piplines rather then the
+  machine readable `Pipeline.spec.tasks.*.name` field.
+* Add a `Pipeline.spec.tasks.*.description` (optional) (or similar) field that allows for a
+  long form description of a task in the context of a `Pipeline` to provide additional
+  human context to the intent of a task in a Pipeline.
```

Comment:
What would be the difference / added value between a `Pipeline.spec.tasks.*.description` and the `Task.spec.description` already available ? As what would we add here that is not present in `Task.spec.description` ?

Reply:
My hunch is that a Task's description will always need to be more general - e.g. `id-generator generates a unique id`.

A PipelineTask's description can be more focused on the Task's purpose _in the pipeline_ - e.g. `generate the id that other tasks in this pipeline will use to tag the built image, name the bucket directory we upload to, and label the deployment pushed to staging`.  Something like that?  This is definitely more human understandable (going by the motivation in the TEP) I think?

Reply:
Precisely. Think of the “git-clone” Task. In the context of a pipeline I may use it multiple times and want to give it multiple different display names. Like. “Clone app code”, “clone app config”, “clone deployment automation”. Just an example, but I think you all get the idea. The purpose of a Task, and thus its name and display name are co to gent on its specific use within a Pipeline.